src/main/java/me/karakelley/http/FileSystem/FileFinderProxy.java

```diff
@@ -0,0 +1,22 @@
+package me.karakelley.http.FileSystem;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FileFinderProxy implements FileFinder {
```

Comment:
I think adding `Proxy` to this name is probaby good/helfpul, but it's also really vague. What functionality does it provide? Does it provide authorization? Caching? Logging?

Note the [`HttpMessageConverter`](https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/http/converter/HttpMessageConverter.html) interface from the Spring project. Notice how all the implementations describe what functionality they provide.